packages/main/src/plugin/telemetry/telemetry.ts

```diff
@@ -275,25 +292,45 @@ export class Telemetry {
 
   // return true if the event needs to be dropped
   protected shouldDropEvent(eventName: string): boolean {
-    // if event is a list event (start with 'list'), do not send it more than one per day
-    if (eventName.startsWith('list')) {
-      // do we have an existing event with the same name?
-      const previousTime = this.lastTimeEvents.get(eventName);
-      // it was not there, so we can send it
-      if (!previousTime) {
-        this.lastTimeEvents.set(eventName, Date.now());
-        return false;
-      }
-      // it was there, so we check if it was more than 24h ago
-      const now = Date.now();
-      const diff = now - previousTime;
-      if (diff > 24 * 60 * 60 * 1000) {
-        this.lastTimeEvents.set(eventName, now);
-        return false;
-      }
+    const telem = this.getTelemetrySettings();
+    if (!telem || !eventName) {
       return true;
     }
-    return false;
+
+    let dropIt = false;
+    telem.forEach(entry => {
+      if (eventName.startsWith(entry.event)) {
+        if (entry.disabled) {
+          // telemetry is entirely disabled for this event
+          dropIt = true;
+        }
+        if (entry.ratio && entry.ratio < 1 && Math.random() > entry.ratio) {
+          // if a ratio is specified, we randomly drop
+          dropIt = true;
+        }
+        if (entry?.frequency === 'dailyPerInstance') {
+          // only send this event once per day, per running instance (not saved to disk)
+          const previousTime = this.lastTimeEvents.get(eventName);
+          // it was not there, so we can send it
+          if (!previousTime) {
+            this.lastTimeEvents.set(eventName, Date.now());
+            return false;
+          } else {
+            // it was there, so we check if it was more than 24h ago
+            const now = Date.now();
+            const diff = now - previousTime;
+            if (diff > 24 * 60 * 60 * 1000) {
+              this.lastTimeEvents.set(eventName, now);
+            } else {
+              dropIt = true;
+            }
+          }
+        }
+      }
+    });
+
+    console.log('telem: ' + eventName + ' ' + dropIt + ' ' + telem.length);
```

Comment:
I think we should remove this

Reply:
Yes, debug statement I meant to remove. Gone now.